community/bisect.go

```diff
@@ -0,0 +1,216 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package community
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+
+	"github.com/gonum/graph"
+)
+
+// Domain is a domain of resolutions with a common score.
+type Domain struct {
```

Comment:
Could you rename this to `Interval`? Domain can mean many things depending on your background. For me, it is an open connected subset of R^n in which you solve a PDE.